Open room member avatar in a media viewer

jmartinesp opened the pull request:
<!-- Please read [CONTRIBUTING.md](https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md) before submitting your pull request -->
 
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

- Moves `MediaViewer` associated code to `:libraries:mediaviewer`, from `:features:messages`.
- Reuses this viewer to display the room member avatar.

## Motivation and context

Closes #1907.

## Tests

- Get into a room, select a user so its details screen is displayed.
- Tap on the avatar in this screen, it should open a media viewer.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 11

## Checklist

<!-- Depending on the Pull Request content, it can be acceptable if some of the following checkboxes stay unchecked. -->

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

bmarty commented on features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/model/event/TimelineItemAudioContent.kt:
```diff
@@ -29,6 +29,10 @@ data class TimelineItemAudioContent(
 
-    val fileExtensionAndSize = formatFileExtensionAndSize(fileExtension, formattedFileSize)
+    val fileExtensionAndSize =
+        io.element.android.libraries.mediaviewer.api.helper.formatFileExtensionAndSize(
```
Maybe cleanup?

bmarty commented on features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/timeline/factories/event/TimelineItemContentMessageFactoryTest.kt:
```diff
@@ -496,7 +496,7 @@ class TimelineItemContentMessageFactoryTest {
     ) = TimelineItemContentMessageFactory(
         fileSizeFormatter = FakeFileSizeFormatter(),
-        fileExtensionExtractor = FileExtensionExtractorWithoutValidation(),
+        fileExtensionExtractor = io.element.android.libraries.mediaviewer.api.util.FileExtensionExtractorWithoutValidation(),
```
Same

bmarty commented on features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsFlowNode.kt:
```diff
@@ -136,9 +148,31 @@ class RoomDetailsFlowNode @AssistedInject constructor(
+                    mediaSource = MediaSource(url = navTarget.avatarUrl),
+                    thumbnailSource = MediaSource(url = navTarget.avatarUrl),
+                    canDownload = false,
+                    canShare = false,
```
Why restricting sharing and downloading here?

bmarty commented on features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/members/details/avatar/RoomMemberAvatarPreviewNode.kt:
```diff
@@ -0,0 +1,31 @@
+    @Assisted buildContext: BuildContext,
+    @Assisted plugins: List<Plugin>,
+    presenterFactory: io.element.android.libraries.mediaviewer.api.viewer.MediaViewerPresenter.Factory,
+) : io.element.android.libraries.mediaviewer.api.viewer.MediaViewerNode(buildContext, plugins, presenterFactory)
```
clean up?

bmarty reviewed (approved):
Thanks for doing this, maybe do some cleanup before merging?

jmartinesp commented:
Thanks for finding these fully qualified usages. I moved lots of classes around so AS broke the imports, and it seems like I only found and fixed some of those.
